packages/odyssey-react-mui/src/theme/useContrastMode.test.tsx

```diff
@@ -0,0 +1,254 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import { render, waitFor, act } from "@testing-library/react";
+import {
+  ContrastModeContext,
+  useContrastMode,
+  useContrastModeContext,
+  getBackgroundColor,
+  hueNeutral50Rgb,
+  type ContrastModeContextType,
+} from "../useContrastMode";
+import * as Tokens from "@okta/odyssey-design-tokens";
+import { renderHook } from "@testing-library/react";
+
+// Common wrapper component for tests
+const createWrapper = (contextValue: ContrastModeContextType) => {
+  return ({ children }: { children: React.ReactNode }) => (
+    <ContrastModeContext.Provider value={contextValue}>
+      {children}
+    </ContrastModeContext.Provider>
+  );
+};
+
+describe("useContrastMode and related functions", () => {
+  const originalGetComputedStyle = window.getComputedStyle;
+
+  beforeEach(() => {
+    window.getComputedStyle = jest.fn().mockImplementation(() => ({
+      backgroundColor: "rgba(0, 0, 0, 0)",
+    }));
+  });
+
+  afterEach(() => {
+    window.getComputedStyle = originalGetComputedStyle;
+    jest.clearAllMocks();
+    document.documentElement.style.backgroundColor = "";
+  });
+
+  describe("useContrastMode hook", () => {
+    it("should return lowContrast mode by default", () => {
+      const { result } = renderHook(() => useContrastMode({}), {
+        wrapper: createWrapper({ contrastMode: "lowContrast" }),
+      });
+
+      expect(result.current.contrastMode).toBe("lowContrast");
+      expect(result.current.contrastContainerRef.current).toBe(null);
+      expect(result.current.parentBackgroundColor).toBe("#ffffff");
+    });
+
+    it("should respect explicitly set contrast mode", () => {
+      const { result } = renderHook(
+        () => useContrastMode({ contrastMode: "highContrast" }),
+        {
+          wrapper: createWrapper({ contrastMode: "lowContrast" }),
+        },
+      );
+
+      expect(result.current.contrastMode).toBe("highContrast");
+    });
+
+    it("should update contrast mode based on background color changes", async () => {
+      const TestComponent = () => {
+        const { contrastContainerRef, contrastMode } = useContrastMode({});
+        return (
+          <div ref={contrastContainerRef} data-testid="container">
+            {contrastMode}
+          </div>
+        );
+      };
+
```

Comment:
This test should probably be in 2 places. Once in `OdysseyThemeProvider`, and once in `getParentBackgroundColor` as this hook itself doesn't care nor does it know about `getComputedStyle` being used.